src/engine/bank02.asm

```diff
@@ -1,17 +1,1775 @@
+	ld d, [hl]
+	inc hl
+	ld e, [hl]
+	inc hl
+
+; text
+	push hl
+	push bc
+	call InitTextPrinting
+	ldtx hl, HandText2
+	call ProcessTextFromID
+	pop bc
+
+; decimal value
+	ld a, b
+	call CalculateOnesAndTensDigits
+	ld hl, wOnesAndTensPlace
+	ld a, [hli]
+	ld b, a
+	ld a, [hl]
+
+	ld hl, wDefaultText
+	ld [hl], TX_SYMBOL
+	inc hl
+	ld [hl], SYM_CROSS
+	inc hl
+	ld [hl], TX_SYMBOL
+	inc hl
+	ld [hli], a
+	ld [hl], TX_SYMBOL
+	inc hl
+
+; draw to screen
+	ld a, b
+	ld [hli], a
+	ld [hl], TX_END
+	ld hl, wDefaultText
+	call ProcessText
+	pop hl
+	ret
+
+; handle player input in menu in Opp. Play Area
+; works out which cursor coordinate to go to
+; and sets carry flag if A or B are pressed
+; input
+; returns a =  $1 if A pressed
+; returns a = $ff if B pressed
+HandleDuelMenuInput_PlayArea: ; 86ac (2:46ac)
+	xor a
+	ld [wcfe3], a
+	ld a, [wCursorDuelXPosition]
+	ld d, a
+	ld a, [wCursorDuelYPosition]
+	ld e, a
+
+; d = cursor x position
+; e = cursor y position
+
+	ldh a, [hDPadHeld]
+	or a
+	jr z, .skip
+
+; pad is pressed
+	ld a, [wce5e]
+	and %10000000
+	ldh a, [hDPadHeld]
+	jr nz, .check_vertical
+	bit D_LEFT_F, a ; test left button
+	jr nz, .horizontal
+	bit D_RIGHT_F, a ; test right button
+	jr z, .check_vertical
+
+; handle horizontal input
+.horizontal
+	ld a, [wce5e]
+	and %01111111
+	or a
+	jr nz, .asm_86dd ; jump if wce5e's lower 7 bits aren't set
+	ld a, e
+	or a
+	jr z, .flip_x ; jump if y is 0
+
+; wce5e = %10000000
+; e = 1
+	dec e ; change y position
+	jr .flip_x
+
+.asm_86dd
+	ld a, e
+	or a
+	jr nz, .flip_x ; jump if y is not 0
+	inc e ; change y position
+.flip_x
+	ld a, d
+	xor $01 ; flip x position
+	ld d, a
+	jr .erase
+
+.check_vertical
+	bit D_UP_F, a
+	jr nz, .vertical
+	bit D_DOWN_F, a
+	jr z, .skip
+
+; handle vertical input
+.vertical
+	ld a, d
+	or a
+	jr z, .flip_y ; jump if x is 0
+	dec d
+.flip_y
+	ld a, e
+	xor $01 ; flip y position
+	ld e, a
+
+.erase
+	ld a, $01
+	ld [wcfe3], a
+	push de
+	call DrawCursorEmpty_OppPlayArea
+	pop de
+
+;update x and y cursor positions
+	ld a, d
+	ld [wCursorDuelXPosition], a
+	ld a, e
+	ld [wCursorDuelYPosition], a
+
+; reset cursor blink
+	xor a
+	ld [wDuelCursorBlinkCounter], a
+
+.skip
+	ldh a, [hKeysPressed]
+	and A_BUTTON | B_BUTTON
+	jr z, .sfx
+	and A_BUTTON
+	jr nz, .a_pressed
+
+; B pressed
+	ld a, $ff ; cancel
+	call PlaySFXConfirmOrCancel
+	scf
+	ret
+	
+.a_pressed
+	call DrawCursor_OppPlayArea
+	ld a, $01
+	call PlaySFXConfirmOrCancel
+	scf
+	ret
+
+.sfx
+	ld a, [wcfe3]
+	or a
+	jr z, .draw_cursor
+	call PlaySFX
+
+.draw_cursor
+	ld hl, wDuelCursorBlinkCounter
+	ld a, [hl]
+	inc [hl]
+	and %00001111
+	ret nz ; only update cursor if blink's lower nibble is 0
+
+	ld a, $0f ; cursor byte
+	bit 4, [hl] ; only draw cursor if blink counter's fourth bit is not set
+	jr z, DrawByteInCursor_OppPlayArea
+; fallthrough
+
+; transforms cursor position into coordinates
+; in order to draw byte on menu cursor
+DrawCursorEmpty_OppPlayArea: ; 8741 (2:4741)
```

Comment:
This is a follow up to the above comment (line 1205). I would replace this one and the function below with ``EraseCheckMenuCursor_YourOrOppPlayArea`` and ``DrawCheckMenuCursor_YourOrOppPlayArea`` respectively. In line with my line 2288 suggestion.